### PR TITLE
Fix mentor form

### PR DESCRIPTION
**Pull request by Isha988**
## Description of Changes

1. Added user name checking functionality and default user name creation functionality.
2. Changed social input
3. Add default session and schedule cards with disclaimer

### CODE REVIEW

**Comment by Isha988** [comment]
I am not able to validate time inputs in schedule section. please suggest me something @Illaf @anmode .

File: components/MentorRegistration/components/ContactDetails.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ function ContactDetails({
       validation: "required|alpha_num_dash_space",
     },
     {
-      label: "linkedIn",
+      label: "linkedIn Handle",
       type: "url",
```
Comment (anmode) [inline components/MentorRegistration/components/ContactDetails.js:39]: handle ->  username
and type will be string

File: components/MentorRegistration/components/ContactDetails.js [hunk 2]
```diff
@@ -48,15 +48,16 @@ function ContactDetails({
...
-          regex:
-            "^https://(www.)?linkedin.com/((in/[^/]+/?)|(pub/[^/]+/((w|d)+/?){3}))$",
+          regex: "[a-z, A-Z]+[0-9,a-z,A-Z,-]*",
         },
       ],
     },
     {
-      label: "twitter",
+      label: "twitter Handle",
       type: "url",
```
Comment (anmode) [inline components/MentorRegistration/components/ContactDetails.js:61]: same here

File: components/MentorRegistration/components/PersonDetails.js [hunk 3]
```diff
@@ -1,14 +1,58 @@
...
+  };
+
+  // onblur function for name input
+  const onNameBlur = (e) => {
+    const name = e.target.value.trim().replaceAll(" ", "-");
+    const randomNumber = window.crypto.getRandomValues(new Uint32Array(1))[0];
+    const userName = `${name}-${randomNumber}`;
+    setFormData({ ...formData, username: userName });
+    checkUserNameAvailability(userName);
+  };
```
Comment (anmode) [inline components/MentorRegistration/components/PersonDetails.js:48]: please move this to components folder

**Review by anmode — COMMENTED** [review]
For validation on schedule side:
1. You have input value so convert the start time to 24 hours ( there is a library for it)
ya dekho this should work: 

```
function validateTimeFormat(timeStr) {
    const regex = /^(0?[1-9]|1[0-2]):[0-5][0-9] [APap][Mm]$/;
    return regex.test(timeStr);
}

function isEndTimeAfterStartTime(startTime, endTime) {
    if (!validateTimeFormat(startTime) || !validateTimeFormat(endTime)) {
        return false; // Invalid time format
    }

    const startDateTime = new Date(`2000-01-01 ${startTime}`);
    const endDateTime = new Date(`2000-01-01 ${endTime}`);

    return endDateTime > startDateTime;
}
```
isEndTimeAfterStartTime(startTime, endTime)

File: components/MentorRegistration/components/PersonDetails.js (hunk 3 above)
Comment (Isha988) [reply components/MentorRegistration/components/PersonDetails.js:48]: Pr ye tho sirf ek function hai

Comment (anmode) [reply components/MentorRegistration/components/PersonDetails.js:48]: ahh... theek h!

**Comment by Isha988** [comment]
@anmode have a look at changes made.

**Comment by Isha988** [comment]
> For validation on schedule side:
> 
>     1. You have input value so convert the start time to 24 hours ( there is a library for it)
>        ya dekho this should work:
> 
> 
> ```
> function validateTimeFormat(timeStr) {
>     const regex = /^(0?[1-9]|1[0-2]):[0-5][0-9] [APap][Mm]$/;
>     return regex.test(timeStr);
> }
> 
> function isEndTimeAfterStartTime(startTime, endTime) {
>     if (!validateTimeFormat(startTime) || !validateTimeFormat(endTime)) {
>         return false; // Invalid time format
>     }
> 
>     const startDateTime = new Date(`2000-01-01 ${startTime}`);
>     const endDateTime = new Date(`2000-01-01 ${endTime}`);
> 
>     return endDateTime > startDateTime;
> }
> ```
> 
> isEndTimeAfterStartTime(startTime, endTime)

Thank you for this, it really helped me a lot.

**Comment by anmode** [comment]
Wow cool,
Just last some change: sorry just click hua
- Make timezone default value as in india ( abhi pacfic aati h)
- Make sesssion type to a drop down
- Session Name also can be drop down ( 1-1 Mentorship , Resume Review , edit ( create your session name))

**Comment by Isha988** [comment]
> * Make sesssion type to a drop down

Dropdown options?

**Comment by Isha988** [comment]
Working on session type input.

**Comment by anmode** [comment]
Wow Isha cool!

**Pull request merged**